### PR TITLE
[Lang] Make python scope comparison return 1 instead of -1

### LINKED ISSUE #5746: vec3 comparison returns -1 in python scope
To reproduce:

```python
import taichi as ti
ti.init()

v = ti.math.vec3(0, 1, 2)
print(v < 1)  # [-1, 0, 0]
```

### PR DESCRIPTION
**Pull request by re-xyr**
Related issue = Fixes #5746

### CODE REVIEW

**Review by strongoier — APPROVED** [review]
Thanks!
nit: fix the pylint error

File: python/taichi/lang/matrix.py [hunk 1]
```diff
@@ -898,10 +898,10 @@ def any(self):
             >>> v.any()
             True
         """
-        ret = ops_mod.cmp_ne(self.entries[0], 0)
-        for i in range(1, len(self.entries)):
```
Comment (strongoier) [inline python/taichi/lang/matrix.py:902]: Rename `i` to `e` or something?

Comment (re-xyr) [reply python/taichi/lang/matrix.py:902]: I guess `elem`

Comment (strongoier) [reply python/taichi/lang/matrix.py:902]: Maybe `entry` is better suited in this file

Comment (re-xyr) [reply python/taichi/lang/matrix.py:902]: You're right

**Pull request merged**